packages/cli/src/command_middleware.ts

```diff
@@ -3,13 +3,19 @@ import { fromNodeProviderChain } from '@aws-sdk/credential-providers';
 import { loadConfig } from '@smithy/node-config-provider';
 import { NODE_REGION_CONFIG_OPTIONS } from '@aws-sdk/region-config-resolver';
 import { AmplifyUserError } from '@aws-amplify/platform-core';
+import { Printer } from '@aws-amplify/cli-core';
 
 export const profileSetupInstruction = `To configure a new Amplify profile, use "npx amplify configure profile".`;
 
 /**
  * Contains middleware functions.
  */
 export class CommandMiddleware {
+  /**
+   * Inject the printer instance so that it can be tested
```

Comment:
```suggestion
   * Creates command middleware.
```

Reply:
For the record... not a huge fan of mandatory comments like these when the code self documents.

Reply:
I don't like them either. But we didn't find a way (or didn't try to find it) to be more precise about where we require them, so now it's necessary evil.